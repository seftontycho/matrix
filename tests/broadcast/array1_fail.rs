#![feature(generic_const_exprs)]

use matrix::prelude::*;

fn main() {
    // impossible shape combinations are:
    // [A], [B] where A != B and A != 1 and B != 1

    array1_array1_unequal();
}

/// This is case: [A], [B] where A != B and A != 1 and B != 1
fn array1_array1_unequal() {
    let a = Array1::from([1, 2, 3]);
    let b = Array1::from([1, 2]);
    let op: fn(&i32, &i32) -> i32 = |a, b| a * b;

    let c = a.broadcast(b, op);
}
