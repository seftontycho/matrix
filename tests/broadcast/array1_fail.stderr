error[E0631]: type mismatch in function arguments
  --> tests/broadcast/array1_fail.rs:18:28
   |
18 |     let c = a.broadcast(b, op);
   |               ---------    ^^
   |               |            |
   |               |            expected due to this
   |               |            found signature defined here
   |               required by a bound introduced by this call
   |
   = note: expected function signature `for<'a, 'b> fn(&'a _, &'b matrix::prelude::Array1<{integer}, 2>) -> _`
              found function signature `for<'a, 'b> fn(&'a _, &'b i32) -> _`
note: required by a bound in `broadcast`
  --> src/broadcast.rs
   |
   |     fn broadcast(self, rhs: Rhs, op: impl Fn(&T, &U) -> V) -> Self::Output;
   |                                           ^^^^^^^^^^^^^^^ required by this bound in `Broadcast::broadcast`
help: consider wrapping the function in a closure
   |
18 |     let c = a.broadcast(b, |arg0: &i32, arg1| op(arg0, /* &i32 */));
   |                            ++++++++++++++++++   ++++++++++++++++++
